parlai/tasks/self_chat/worlds.py

```diff
@@ -5,51 +5,81 @@
 # LICENSE file in the root directory of this source tree.
 
 import random
+from typing import Any, Dict, List
 
+from parlai.core.agents import Agent
 from parlai.core.worlds import DialogPartnerWorld, validate
 
 
-class InteractiveWorld(DialogPartnerWorld):
+class SelfChatBaseWorld(DialogPartnerWorld):
     def __init__(self, opt, agents, shared=None):
         super().__init__(opt, agents, shared)
         self.init_contexts()
-        self.max_cnt = self.opt.get('selfchat_max_turns', 10)
-        self.cnt = 0
+        self.init_openers()
+        self.max_turn_cnt = self.opt.get('selfchat_max_turns', 10)
+        self.turn_cnt = 0
+        self.episode_cnt = 0
 
     def init_contexts(self):
         pass
 
-    def get_contexts(self):
+    def get_contexts(self, episode_num: int) -> List[str]:
         return ['__SILENCE__', '']
 
+    def init_openers(self) -> None:
+        pass
+
+    def get_openers(self, episode_num: int) -> List[str]:
+        pass
+
     def display(self):
         s = ''
         s += super().display()
-        if self.cnt == 0:
+        if self.turn_cnt == 0:
             s += '\n==============================\n'
         return s
 
     def episode_done(self):
-        if self.cnt > self.max_cnt:
+        if self.turn_cnt >= self.max_turn_cnt:
             return True
         else:
             return False
 
+    def _get_seed_utt_acts(
+        self, episode_num: int, agents: List[Agent]
+    ) -> List[Dict[str, Any]]:
+        def make_agent_action(utterance: str, agent: Agent) -> Dict[str, Any]:
+            return {'text': utterance, 'episode_done': False, 'id': agent.id}
+
+        openers = self.get_openers(episode_num)
```

Comment:
how do you pass in openers?

Reply:
should this somehow be related to the task? can we find a way to construct them automatically

Reply:
Yeah, that's a great idea! I'll add a flag to optionally just grab a random first message from the dataset.